### DISCUSSION TITLE
Add element index settings to project config

### DISCUSSION THREAD

**Discussion by carlcs** [craftcms/cms#8616] [Ideas] [answered]
According to your [check list](https://docs.craftcms.com/v3/extend/project-config.html#is-project-config-support-right-for-you) it’s not a bad candidate at least.

**Comment by brandonkelly** [comment]
Yeah agreed.

**Comment by jsunsawyer** [comment]
Digging into Project Config right now and came across this when looking for gotchas.

Might be nice if the docs clearly stated everything that's _not_ stored in the project config. Right now it's just a list of everything that's included, and one would assume that it's everything that _should_ be included. https://docs.craftcms.com/v3/project-config.html

**Comment by brandonkelly** [comment]
> and one would assume that it's everything that _should_ be included

That’s the case.

We haven’t added this yet because we decided that element index settings actually shouldn’t be, because eventually each user is going to be able to customize element indexes to their own liking.

**Comment by benface** [comment]
> We haven’t added this yet because we decided that element index settings actually shouldn’t be, because eventually each user is going to be able to customize element indexes to their own liking.

That's cool, but I still think the "default" element indexes (when a user doesn't care, or before they customize them) should be stored in project config.

**Comment by brandonkelly** [comment]
Saving defaults in project config makes sense.

**Comment by benface** [comment]
Is this a 3.x thing or probably 4.0? Project config has definitely spoiled me, but it's slightly annoying to have to go in the production site's CP and manually adjust some settings after a deployment, especially when everything else just syncs automatically.

**Comment by brandonkelly** [comment, marked as answer]
We are going to be making some big changes to element indexes in 4.0, so it makes sense to wait until then.

**Reply by brandonkelly** [reply]
For posterity, this was in fact added in 4.0 :)

**Comment by benface** [comment]
That's what I thought. Thanks!

**Comment by mmikkel** [comment]
This sounds great, just wanted to emphasise the importance of the developer still being able to configure default index settings, even if each user can override those defaults with their own (and yes, those defaults should be stored in PC) 👍

**Comment by brandonkelly** [comment]
For sure.

**Comment by sarahschuetz** [comment]
Any update on this? :)

**Reply by mmikkel** [reply]
It was added in Craft 4.0.
